src/BenchmarkDotNet/Running/BenchmarkRunnerClean.cs

```diff
@@ -531,13 +532,26 @@ private static BenchmarkReport RunCore(BenchmarkCase benchmarkCase, BenchmarkId
         private static void LogTotalTime(ILogger logger, TimeSpan time, int executedBenchmarksCount, string message = "Total time")
             => logger.WriteLineStatistic($"{message}: {time.ToFormattedTotalTime(DefaultCultureInfo.Instance)}, executed benchmarks: {executedBenchmarksCount}");
 
-        private static BenchmarkRunInfo[] GetSupportedBenchmarks(BenchmarkRunInfo[] benchmarkRunInfos, ILogger logger, IResolver resolver)
-            => benchmarkRunInfos.Select(info => new BenchmarkRunInfo(
-                    info.BenchmarksCases.Where(benchmark => benchmark.GetToolchain().IsSupported(benchmark, logger, resolver)).ToArray(),
+        private static BenchmarkRunInfo[] GetSupportedBenchmarks(BenchmarkRunInfo[] benchmarkRunInfos, ILogger logger, IResolver resolver, out List<ValidationError> validationErrors)
```

Comment:
inside `BenchmarkRunnerClean` it's OK to use `ValueTuples`, we could use them here as well:

```suggestion
        private static (BenchmarkRunInfo[] benchmarks, IReadOnlyList<ValidationError> validationErrors) GetSupportedBenchmarks(BenchmarkRunInfo[] benchmarkRunInfos, ILogger logger, IResolver resolver)
```